File: Python/1047 - Game Time with Minutes.py
initial_hour, initial_minute, final_hour, final_minute = map(int, input().split())

if initial_hour == final_hour and initial_minute == final_minute:
  hours_time = 24
  minutes_time = 0
else:
  minutes_time = final_minute - initial_minute
  if initial_hour < final_hour or (initial_hour == final_hour and minutes_time >= 0):
    hours_time = final_hour - initial_hour
  else:
    hours_time = 24 - initial_hour + final_hour

  if minutes_time < 0:
    minutes_time += 60
    hours_time -= 1

print("O JOGO DUROU", hours_time, "HORA(S) E", minutes_time, "MINUTO(S)")

File: Java/1047 - Game Time with Minutes.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int initialHour = sc.nextInt();
        int initialMinute = sc.nextInt();
        int finalHour = sc.nextInt();
        int finalMinute = sc.nextInt();
        int hoursTime, minutesTime;

        if (initialHour == finalHour && initialMinute == finalMinute) {
            hoursTime = 24;
            minutesTime = 0;
        } else {
            minutesTime = finalMinute - initialMinute;
            if (initialHour < finalHour) {
                hoursTime = finalHour - initialHour;
            } else if (initialHour == finalHour) {
                if (minutesTime >= 0) {
                    hoursTime = 0;
                } else {
                    hoursTime = 24 - initialHour + finalHour;
                }
            } else {
                hoursTime = 24 - initialHour + finalHour;
            }

            if (minutesTime < 0) {
                minutesTime += 60;
                hoursTime -= 1;
            }
        }

        System.out.printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", hoursTime, minutesTime);
    }
}
